src/libmongoc/src/mongoc/mongoc-uri.c

```diff
@@ -1466,10 +1492,14 @@ mongoc_uri_parse (mongoc_uri_t *uri, const char *str, bson_error_t *error)
    }
 
    bson_free (before_slash);
+   bson_free (userpass);
+   bson_free (hostname);
```

Comment:
`userpass` and `hostname` are not used outside of the "Handle cases of optional delimiting slash" block, and there are no jumps out of the block. Move the `bson_free()` calls to the end of that block + reduce scope of the `userpass` and `hostname` variables.

```c
if (!before_slash) {
   // Handle cases of optional delimiting slash

   const char* const userpass = ...;
   ...

   bson_free (userpass);
   bson_free (hostname);
}
```